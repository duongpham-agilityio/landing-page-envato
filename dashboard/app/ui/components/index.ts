export { default as InputField } from './common/InputField';
export { default as FallbackErrorBoundary } from './FallbackErrorBoundary';
export { default as Navigation } from './common/Navigation';
export { default as Menu } from './common/Menu';
export { default as Logo } from './common/Logo';
export { default as ExpandSidebar } from './ExpandSidebar';
export { default as MiniSidebar } from './MiniSidebar';
export * from './Icons';
export * from './Images';
export { default as Avatar } from './common/Avatar';
export { default as Benefit } from './common/Benefit';
export { default as Divider } from './common/Divider';
export { default as IconButton } from './common/IconButton';
export { default as Dropdown } from './common/Dropdown';
export { default as Select } from './common/Select';
export { default as Pagination } from './common/Pagination';
export { default as Button } from './common/Button';
export { default as Table } from './common/Table';
export { default as CustomerNameCell } from './common/Table/Body/CustomerNameCell';
export { default as ActionCell } from './common/Table/Body/ActionCell';
export { default as UserInfoCell } from './UsersTable/Cell/UserInfoCell';
export { default as StatusCell } from './UsersTable/Cell/StatusCell';
export { default as HeadCell } from './common/Table/HeadCell';
export { default as Selector } from './TransactionTable/Selector';
export { default as TransactionTable } from './TransactionTable';
export { default as SearchBar } from './common/SearchBar';
export { default as SwitchTheme } from './SwitchTheme';
export { default as CardPayment } from './CardPayment';
export { default as BoxChat } from './BoxChat';
export { default as TotalStatisticCard } from './TotalStatisticCard';
export { default as Fetching } from './Skeleton/Fetching';
export { default as TotalStatisticListSkeleton } from './Skeleton/TotalStatisticListSkeleton';
export { default as TotalStatisticList } from './TotalStatisticList';
export { default as RevenueFlow } from './RevenueFlow';
export { default as Efficiency } from './Efficiency';
export { default as TotalBalance } from './TotalBalance';
export { default as OverallBalance } from './OverallBalance';
export { default as UserCard } from './UserCard';
export { default as ItemSideBarSetting } from './ItemSideBarSetting';
export { default as FaqItem } from './FaqItem';
export { default as UpdateProfile } from './Profile';
export { default as AuthForm } from './AuthForm';
export { default as Modal } from './common/Modal';
export { default as FallbackImage } from './FallbackImage';
export { default as ChatMember } from './ChatMember';
export { default as Conversation } from './Conversation';
export { default as PinCode } from './common/PinCode';
