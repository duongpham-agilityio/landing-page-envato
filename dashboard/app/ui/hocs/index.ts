export { default as withLazy } from './withLazy';
export * from './withErrorBoundary';
export * from './withAuthentication';
