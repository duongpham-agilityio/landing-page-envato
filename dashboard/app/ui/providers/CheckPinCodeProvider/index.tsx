'use client';

import dynamic from 'next/dynamic';
import { useCallback, useMemo } from 'react';
import { useDisclosure, useToast } from '@chakra-ui/react';
import { getMessaging, onMessage } from 'firebase/messaging';
import { SubmitHandler, useForm } from 'react-hook-form';
import { useQueryClient } from '@tanstack/react-query';

//Constants
import {
  END_POINTS,
  ERROR_MESSAGES,
  SHOW_TIME,
  STATUS,
  SUCCESS_MESSAGES,
} from '@/lib/constants';

// Hooks
import { useAuth, usePinCode } from '@/lib/hooks';

// Types
import { TPinCodeForm } from '@/lib/interfaces';

// Stores
import { TAuthStoreData, authStore } from '@/lib/stores';

// Utils
import { customToast, isWindowDefined } from '@/lib/utils';

const Modal = dynamic(() => import('@/ui/components/common/Modal'));
const PinCode = dynamic(() => import('@/ui/components/common/PinCode'));

const CheckPinCodeProvider = () => {
  const user = authStore((state): TAuthStoreData['user'] => state.user);

  const { isOpen: isPinCodeModalOpen, onClose: onClosePinCodeModal } =
    useDisclosure({ isOpen: true });

  const {
    control,
    handleSubmit,
    formState: { isValid, isSubmitting },
  } = useForm<TPinCodeForm>({
    defaultValues: {
      pinCode: '',
    },
  });

  const { setUser } = useAuth();

  const toast = useToast();

  const { handleSetPinCode } = usePinCode();

  const onSubmitPinCode: SubmitHandler<TPinCodeForm> = useCallback(
    (data) => {
      if (user) {
        data.userId = user.id;

        try {
          handleSetPinCode(data);

          setUser({ user: { ...user, pinCode: data.pinCode } });

          onClosePinCodeModal();

          toast(
            customToast(
              SUCCESS_MESSAGES.SET_PIN_CODE.title,
              SUCCESS_MESSAGES.SET_PIN_CODE.description,
              STATUS.SUCCESS,
            ),
          );
        } catch (error) {
          toast(
            customToast(
              ERROR_MESSAGES.SET_PIN_CODE.title,
              ERROR_MESSAGES.SET_PIN_CODE.description,
              STATUS.ERROR,
            ),
          );
        }
      }
    },
    [handleSetPinCode, onClosePinCodeModal, setUser, toast, user],
  );

  const pinCodeModalBody = useMemo(
    () => (
      <PinCode
        control={control}
        isDisabled={!isValid || isSubmitting}
        onSubmit={handleSubmit(onSubmitPinCode)}
        onClose={onClosePinCodeModal}
      />
    ),
    [
      control,
      handleSubmit,
      isSubmitting,
      isValid,
      onClosePinCodeModal,
      onSubmitPinCode,
    ],
  );

  const queryClient = useQueryClient();

  const messaging = isWindowDefined() ? getMessaging() : null;

  messaging &&
    onMessage(messaging, async (payload) => {
      await Promise.all([
        queryClient.invalidateQueries({
          queryKey: [END_POINTS.MY_WALLET],
        }),
        queryClient.invalidateQueries({
          queryKey: [END_POINTS.TRANSACTIONS],
        }),
        queryClient.invalidateQueries({
          queryKey: [END_POINTS.NOTIFICATION],
        }),
      ]).finally(() => {
        toast({
          title: payload?.notification?.title,
          description: payload?.notification?.body,
          status: 'success',
          duration: SHOW_TIME,
          isClosable: true,
          position: 'top-right',
        });
      });
    });

  return (
    isPinCodeModalOpen && (
      <Modal
        title="Please set the PIN code to your account"
        isOpen={isPinCodeModalOpen}
        onClose={onClosePinCodeModal}
        body={pinCodeModalBody}
      />
    )
  );
};

export default CheckPinCodeProvider;
