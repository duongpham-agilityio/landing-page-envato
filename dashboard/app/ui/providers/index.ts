export { default as ChakraProvider } from './ChakraUI';
export { default as QueryProvider } from './Query';
export { default as CheckAuthenticationProvider } from './CheckAuthentication';
export { default as CheckPinCodeProvider } from './CheckPinCode';
