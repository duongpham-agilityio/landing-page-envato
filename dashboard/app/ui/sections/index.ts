export { default as UsersSection } from './UsersSection';
export { default as SettingsSection } from './SettingsSection';
export { default as SupportsSection } from './SupportSection';
export { default as ForgotPasswordSection } from './ForgotPassword';
export { default as AccountSection } from './AccountSection';
export { default as WalletSection } from './WalletSection';
