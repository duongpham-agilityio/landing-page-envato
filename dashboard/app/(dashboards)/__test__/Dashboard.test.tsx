import { QueryProvider } from '@/ui/providers';
import Dashboard from '../page';
import { useSearchParams } from 'next/navigation';

describe('Dashboard render', () => {
  beforeAll(async () => {
    (useSearchParams as jest.Mock).mockReturnValue({
      get: jest.fn(),
      forEach: jest.fn(),
    });
  });

  test('Should render match with snapshot.', async () => {
    const { container } = render(await Dashboard(), { wrapper: QueryProvider });
    await waitFor(() => {
      expect(container).toMatchSnapshot();
    });
  });
});
