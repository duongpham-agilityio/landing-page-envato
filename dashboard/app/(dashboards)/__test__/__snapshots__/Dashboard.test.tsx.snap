// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard render Should render match with snapshot. 1`] = `
<div>
  <div
    class="css-14tf0fk"
  >
    <div
      class="css-rklm6r"
    >
      <div
        class="css-ny25bk"
      >
        <div
          class="css-1b0f144"
        />
        <div
          class="css-1myajjl"
        />
      </div>
      <section
        class="css-1f9kltt"
      >
         
      </section>
    </div>
    <div
      class="css-r2mz9g"
    >
      <div
        class="chakra-stack css-txpvqp"
      >
        <div
          class="css-1ki54i"
        />
        <div
          class="css-1o7vdud"
        />
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
